### ISSUE TITLE
anyone see this compilation error

### ISSUE THREAD

**Issue by GitPaean** [OPM/opm-grid#226]
Parallel buiding, with either dune 2.3.1 or dune 2.4.1. Ubuntu 16.04 with gcc version 5.4.

```
CMakeFiles/grid_test.dir/tests/grid_test.cc.o: In function `Dune::VTKWriter<Dune::GridView<Dune::DefaultLeafGridViewTraits<Dune::CpGrid, (Dune::PartitionIteratorType)4> > >::pwrite(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Dune::VTK::OutputType, int, int)':
/home/kaib/local/dune2.4.1/release2.4.1_O2_g/include/dune/grid/io/file/vtk/vtkwriter.hh:927: undefined reference to `Dune::concatPaths(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/home/kaib/local/dune2.4.1/release2.4.1_O2_g/include/dune/grid/io/file/vtk/vtkwriter.hh:928: undefined reference to `Dune::relativePath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/grid_test.dir/tests/grid_test.cc.o: In function `Dune::VTKWriter<Dune::GridView<Dune::DefaultLeafGridViewTraits<Dune::CpGrid, (Dune::PartitionIteratorType)4> > >::getSerialPieceName(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
/home/kaib/local/dune2.4.1/release2.4.1_O2_g/include/dune/grid/io/file/vtk/vtkwriter.hh:837: undefined reference to `Dune::concatPaths(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/grid_test.dir/tests/grid_test.cc.o: In function `Dune::VTKWriter<Dune::GridView<Dune::PolyhedralGridViewTraits<3, 3, (Dune::PartitionIteratorType)4> > >::getSerialPieceName(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
/home/kaib/local/dune2.4.1/release2.4.1_O2_g/include/dune/grid/io/file/vtk/vtkwriter.hh:837: undefined reference to `Dune::concatPaths(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
CMakeFiles/grid_test.dir/build.make:125: recipe for target 'bin/grid_test' failed
make[2]: *** [bin/grid_test] Error 1
CMakeFiles/Makefile2:168: recipe for target 'CMakeFiles/grid_test.dir/all' failed
make[1]: *** [CMakeFiles/grid_test.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by GitPaean** [comment]
The prototype of the function `concatPaths` in `path.hh` is 
```
std::string concatPaths(const std::string& base, const std::string& p);
```

**Comment by atgeirr** [comment]
Unable to reproduce on my platform.

**Comment by GitPaean** [comment]
According to this link,  http://stackoverflow.com/questions/33394934/converting-std-cxx11string-to-stdstring

It is possible related to `_GLIBCXX_USE_CXX11_ABI`, while I did not manage to fix the problem by setting the marco to be 1 and recompiling dune.

**Comment by blattms** [comment]
Did you perhaps use different compilers for DUNE and OPM?

**Comment by GitPaean** [comment]
> Did you perhaps use different compilers for DUNE and OPM?
I only have one version of compilers in the system. 

I have the same system on my laptop. The serial building on that laptop worked yesterday. 

So I tried to `dist-upgrade` the system. Beginning from sucessfully serial compilation by removing mpi, recompiled dune 2.4.1, reinstalled Zoltan, try step by step to see which step causes the problem. 

However, all the steps work now, and sorry for not figuring out something helpful. 

The problem from this issue I reproduced for more than 10 times yesterday and today, and dune was recompiled with different version for several times too. I do not know exacty what caused the problem. 

Since the problem kind of goes away now, I will close the issume.
